src/Model/Translatable/TranslatableMethods.php

```diff
@@ -56,6 +56,28 @@ public function addTranslation($translation)
     }
 
     /**
+     * Setter for translations collection
+     * 
+     * @param ArrayCollection|Array $translations
+     *
+     * @return $this
+     */
+    public function setTranslations($translations)
+    {
+        if (!$translations instanceof \Traversable && !is_array($translations)) {
+            throw new \InvalidArgumentException('$translations parameter must be iterable');
+        }
+
+        $this->translations = new ArrayCollection();
```

Comment:
I'm not sure doctrine will like to replace a persistent collection with another collection. That's why I'd like to see an integration test.